core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1552,30 +1560,45 @@ class KafkaApis(val requestChannel: RequestChannel,
         sendResponseMaybeThrottle(request, requestThrottleMs =>
           new AddPartitionsToTxnResponse(requestThrottleMs, partitionErrors.asJava))
       } else {
-        // Send response callback
-        def sendResponseCallback(error: Errors): Unit = {
-          def createResponse(requestThrottleMs: Int): AbstractResponse = {
-            val responseBody: AddPartitionsToTxnResponse = new AddPartitionsToTxnResponse(requestThrottleMs,
-              partitionsToAdd.asScala.map{tp => (tp, error)}.toMap.asJava)
-            trace(s"Completed $transactionalId's AddPartitionsToTxnRequest with partitions $partitionsToAdd: errors: $error from client ${request.header.clientId}")
-            responseBody
+        val (_, incorrectMessageFormat) = partitionsToAdd.asScala.partition { tp =>
+          replicaManager.getMagic(tp) match {
```

Comment:
If we had an easy way to check for unsupported format, then I agree, but if we need to introduce a new path for accessing non-local configs just for this use case, I'd say skip it. Instead of getting the error here, the user will get one when they try to write to the partition and will have to abort the transaction. As long as we expect the error from WriteTxnMarker, no harm done. What do you think?